lib/appium.js

```diff
@@ -60,6 +62,18 @@ class AppiumDriver extends BaseDriver {
     let d = new InnerDriver(this.args);
     let [innerSessionId, dCaps] = await d.createSession(caps, reqCaps, curSessions);
     this.sessions[innerSessionId] = d;
+
+    // Remove the session on unexpected shutdown, so that we are in a position
+    // to open another session later on.
+    // TODO: this should be removed and replaced by a onShutdown callback.
+    d.onUnexpectedShutdown
```

Comment:
why not make use of the unexpected shutdown facility baked into basedriver?

```js
d.onUnexpectedShutdown.catch(err => {
  await this.startUnexpectedShutdown(err);
});
```

Is it too different?